### PR TITLE
W-035944 - Add new custom flow and custom tasks to enable Shield Platform Encryption

### PR DESCRIPTION
**Pull request by robertbwatson**
A new custom flow and three new custom tasks were developed for EDA in order to setup Shield Platform Encryption more easily in a developer scratch org.

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes
1. Create a new developer scratch org.
2. After deploying EDA to the scratch org (using the `dev_org` or `qa_org` flow), run the `setup_encryption` flow.
3. Open the org and navigate to *Setup | Platform Encryption | Encryption Policy | Encrypt Fields*
4. Enable all standard fields for encryption.
    a. Optionally, enable EDA custom fields for encryption.
5. Run the EDA unit tests. Confirm they all pass.

(See Quip doc in WI for more details)

### CODE REVIEW

File: cumulusci.yml [hunk 1]
```diff
@@ -37,6 +37,27 @@ tasks:
...
+        class_path: cumulusci.tasks.sfdx.SFDXOrgTask
+        options:
+            command: 'force:user:permset:assign -n Encryption'
+    
+    create_tenant_secret:
+        group: 'Shield Platform Encryption'
+        description: Creates a new tenant secret for Shield Platform Encryption testing.
+        class_path: cumulusci.tasks.sfdx.SFDXOrgTask
+        options:
+            command: 'force:data:record:create -s TenantSecret -v "Description=Testing"'
```
Comment (robertbwatson) [inline cumulusci.yml]: Got this from https://github.com/wadewegner/sfdx-platformencryption -- the values option was required and for some reason I don't even know where the Description shows up in the UI as I wasn't able to see it on the key record...

File: cumulusci.yml [hunk 2]
```diff
@@ -37,6 +37,27 @@ tasks:
         options:
             path: unpackaged/config/dev_delete
 
+    deploy_encryption_permset:
+        group: 'Shield Platform Encryption'
+        description: Deploys "Manage Encryption Keys" permission set.
+        class_path: cumulusci.tasks.salesforce.Deploy
+        options:
+            path: unpackaged/config/spe
```
Comment (jstvz) [inline cumulusci.yml]:
```suggestion
            path: unpackaged/config/spe
            namespace_inject: $project_config.project__package__namespace
```

We don't have any tokens yet, but generally a good practice to set the project as default.

**Comment by robertbwatson** [comment]
> Looks good, one change requested.
> 
> What's the use case here? Automated testing, or something else? I ask because this flow will require a scratch org configuration with `PlatformEncryption` enabled:
> 
> ```json
> {
>   "orgName": "EDA - Dev Workspace",
>   "edition": "Developer",
>   "features": ["PlatformEncryption"],
>   "settings" : {
>     "orgPreferenceSettings": {
>         "s1DesktopEnabled": true,
>         "translation": true,
>         "chatterEnabled": true
>     }
>   }
> }
> ```
> 
> If we're going to use this for builds, we'll need to include a scratch org definition file that has this feature enabled. If not, we should document this fact somewhere (in the configuration readme, the flow description, or both).

@jstvz Sorry, this got assigned to you maybe a little early. I don't know exactly when this flow will be used, but probably during regression and after developing a feature with a lot of Apex. There's a Quip doc that I'm working on that will go into more detail.

I've been testing these flows and tasks without a special configuration for my developer scratch org. I too thought we would need to enable PlatformEncryption, but that does not appear to be true.

https://releasenotes.docs.salesforce.com/en-us/summer19/release-notes/rn_security_pe_scratch_org.htm -- maybe adding this feature to the scratch org configuration only applies if the org is not defined as a "Developer" edition org?

**Comment by jstvz** [comment]
Nope, you're right; I ran this against `release`. My original comment above has been edited to reflect this.

**Comment by robertbwatson** [comment]
@lifewithryan I'm assigning this to you since I know you were looking over the Quip doc... not sure if you wanted to review this as well (James just approved)

**Comment by ceiroa** [comment]
Let me know if you'd like me to take this one over, @lifewithryan.

**Comment by lifewithryan** [comment]
> Let me know if you'd like me to take this one over, @lifewithryan.

@ceiroa feel free, from what I see I'm good but if you wanna give it  glance that'd be great.

**Comment by ceiroa** [comment]
Hey @robertbwatson, would you mind adding some testing notes to this PR? Just for me and QA to know how to verify it. Thanks!

**Comment by ceiroa** [comment]
@dgamboni One test in the Feature Test build failed, but it was probably just a temporary thing. Rebuilding now. Should be good to go once that passes.

**Pull request merged**